urllib3/util/selectors.py

```diff
@@ -99,7 +104,7 @@ def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
             except (OSError, IOError, select.error) as e:
                 # select.error wasn't a subclass of OSError in the past.
                 errcode = None
-                if hasattr(e, "errno"):
+                if hasattr(e, "errno") and e.errno is not None:
```

Comment:
e.errno is None if OSError or IOError are raised without a second parameter. I don't know if that is done by anything we're calling, but this way if it is we'll fallback to e.args[0]